lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
@danielbachhuber thanks! That's exactly what we need here. So then for the widgets endpoints to be analogous to the post endpoints, we'd want to implement `/wp/v2/widgets/:type` which would expose the schema for that individual widget type, and this endpoint can receive `POST` requests to create new widgets of that type.

So then I don't suppose there is a generic endpoint for getting all posts of all types, other than to add a `filter[post_type]=any` param, but we could allow this for widgets by implementing `/wp/v2/widgets`. Getting all widgets regardless of type would seem to be more commonly needed than getting all posts regardless of type.

I think a `widget` resource would need to have a body that looks like:

```json
{
	"id": "text-123",
	"type": "text",
	"content": {
		"title": "Hello World",
		"text": "Paragraph one\n\nParagraph two",
		"filter": true
	}
}
```

This would be very similar to the `post` resource, where the `content` here is containing the widget instance array.

With the `type` specified here, we would be able to accept `POST` requests to `/wp/v2/widgets` rather than having to `POST` to the endpoint for the specific type, like `/wp/v2/widgets/text`.

Now, for address a single widget instance, I see there being two options:

* `/wp/v2/widgets/:id`
* `/wp/v2/widgets/:type/:id`

The first one has the possibility for ambiguity, if it so happens that a widget type is registered that has an `id_base` that is the same as an `id`, like if you registered a widget type `text-2`. This I should think is very unlikely, and we could specify the pattern in such a `type` (`id_base`) cannot end in `/-\d+$/`. Eventually, if widgets get stored in posts instead of options ([#35669](https://core.trac.wordpress.org/ticket/35669)), then the ID would be a bare integer without an `id_base` part, and so the chance of ambiguity would be even less. In that case we would need to disallow that a `type` (`id_base`) contains _only_ digits. So I suggest the pattern for the `:id` be `^(.+-)?\d+$` and the `:type` parameter be either the somewhat inverse of this, like `^.*\D$`, or to take the list of all registered widget types concatenated together: actually, the latter would be what it should be in practice, as a new endpoints would need to be registered with schema information for each `WP_Widget` instantiated.